src/zenml/client.py

```diff
@@ -1779,6 +1782,44 @@ def delete_stack(self, name_id_or_prefix: Union[str, UUID]) -> None:
                 f"one."
             )
 
+        if recursive:
+
+            stack_components_free_for_deletion = []
+
+            # Get all stack components associated with this stack
+
+            for component_type, component_model in stack.components.items():
+
+                # Get stack associated with the stack component
+
+                stacks = self.list_stacks(
+                    component_id=component_model[0].id, size=2, page=1
+                )
+
+                # Check if the stack component is part of another stack
+                if len(stacks) == 1:
+                    if stack.id == stacks[0].id:
+                        stack_components_free_for_deletion.append(
+                            (component_type, component_model)
+                        )
+
+            self.delete_stack(stack.id)
+
+            for (
+                stack_component_type,
+                stack_component_model,
+            ) in stack_components_free_for_deletion:
+                self.delete_stack_component(
+                    stack_component_model[0].name, stack_component_type
+                )
+                logger.info(
```

Comment:
I think this log is already happening in `self.delete_stack_component(...)` right? If that's the case, I would remove it here.